src/ProgrammableDevice.cc

```diff
@@ -0,0 +1,109 @@
+#include "ProgrammableDevice.hh"
+
+#include "CommandController.hh"
+#include "MSXCliComm.hh"
+#include "MSXCPUInterface.hh"
+#include "MSXException.hh"
+#include "StringSetting.hh"
+#include "serialize.hh"
+
+#include "narrow.hh"
+
+namespace openmsx {
+
+ProgrammableDevice::ProgrammableDevice(const DeviceConfig& config)
+	: MSXDevice(config)
+	, portListSetting(
+		getCommandController(), tmpStrCat(getName(), " ports"),
```

Comment:
Here you use `tmpStrCat(getName(), " ports")` to construct the setting name, but below you use `getName() + "..."` to construct the callback names. I checked: in both cases the constructor expects a `std::string_view` parameter. So both are correct, but we shouldn't mix both approaches for no reason. Pick one and use it for all 4 cases.